src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.Cache.cs

```diff
@@ -0,0 +1,86 @@
+﻿extern alias __uno;
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator;
+
+internal partial class XamlFileParser
+{
+	private struct CachedFileKey : IEquatable<CachedFileKey>
+	{
+		public CachedFileKey(string includeXamlNamespaces, string excludeXamlNamespaces, string file, ImmutableArray<byte> checksum)
+		{
+			IncludeXamlNamespaces = includeXamlNamespaces;
+			ExcludeXamlNamespaces = excludeXamlNamespaces;
```

Comment:
Storing these as `string`s here instead of `string[]` for proper comparison and hash code.